cirq-core/cirq/ops/raw_types.py

```diff
@@ -375,6 +377,55 @@ def _rmul_with_qubits(self, qubits: Tuple['cirq.Qid', ...], other):
     def _json_dict_(self) -> Dict[str, Any]:
         return protocols.obj_to_dict_helper(self, attribute_names=[])
 
+    def on_each(self, *targets: Union[Qid, Iterable[Any]]) -> List['Operation']:
+        """Returns a list of operations applying the gate to all targets.
+
+        Args:
+            *targets: The qubits to apply this gate to. For single-qubit gates
+            this can be provided as varargs or a combination of nested
+            iterables. For multi-qubit gates this must be provided as an
+            `Iterable[Sequence[Qid]]`, where each sequence has `num_qubits`
+            qubits.
+
+        Returns:
+            Operations applying this gate to the target qubits.
+
+        Raises:
+            ValueError if targets are not instances of Qid or Iterable[Qid].
+            ValueError if the gate qubit number is incompatible.
+        """
+        operations: List['Operation'] = []
+        if self._num_qubits_() > 1:
+            if len(targets) != 1 or not isinstance(targets[0], Iterable):
+                raise ValueError(f'The inputs for multi-qubit gates cannot be in varargs form.')
+            for target in targets[0]:
+                if not isinstance(target, Sequence):
+                    if isinstance(target, Qid):
+                        raise ValueError(
+                            f'The inputs for multi-qubit gates cannot be in varargs form.'
+                        )
+                    else:
```

Comment:
for example: you don't need `else` here since the if block returns (via raising an exception)